### ISSUE TITLE
nested json point to the wrong parent

### ISSUE THREAD

**Issue by hanct** [treedoc/treedoc_ts#10]
Hi,
With regards to my raised issue on https://github.com/treedoc/treedoc_ts/issues/9, i think the issue not yet resolved.

To reiterate on what is raised on my previous issue:
All FieldItem objects ' parent are pointing "Root" fieldItem (which is correct), except for "Field 4" and "Field 5" fieldItem instances' whose parent also points to the "Root" (which is incorrect). The "Field 4" and "Field 5" should be pointing to "Group 1" fieldItem instead.
```
this.rootFieldItem= new FieldItem( {fieldName:"Root", fieldType:FIELD_ITEM_TYPE.FIELD_GROUP_ROOT})
this.rootFieldItem.arrChildFieldItem=[
          new FieldItem({fieldName:"Field 1", fieldType:FIELD_ITEM_TYPE.FIELD_ITEM_LEAF, parent:this.rootFieldItem, fieldValue:"aa", fieldSize:5}),
          new FieldItem({fieldName:"Field 2", fieldType:FIELD_ITEM_TYPE.FIELD_ITEM_LEAF, parent:this.rootFieldItem, fieldValue:"bb", fieldSize:6}),
          new FieldItem({fieldName:"Group 1", fieldType:FIELD_ITEM_TYPE.FIELD_GROUP, parent:this.rootFieldItem}),
          new FieldItem({fieldName:"Field 3", fieldType:FIELD_ITEM_TYPE.FIELD_ITEM_LEAF, parent:this.rootFieldItem, fieldValue:"cc", fieldSize:10})
]  
this.rootFieldItem.arrChildFieldItem[2].arrChildFieldItem=[
          new FieldItem({ fieldName:"Field 4", fieldType:FIELD_ITEM_TYPE.FIELD_ITEM_LEAF, parent:this.rootFieldItem.arrChildFieldItem[2],fieldValue:"cc1", fieldSize:5}),
          new FieldItem({fieldName:"Field 5", fieldType:FIELD_ITEM_TYPE.FIELD_ITEM_LEAF, parent:this.rootFieldItem.arrChildFieldItem[2],fieldValue:"cc2f", fieldSize:5})
]
```

Below is how I do the stringify and then parse it to get back the obj. And via this obj , i made the above observation.
I suspect there might need to configure some TD options for that to work. But I could not find any documentation for the options
```
 const strTemplate = TD.stringify(this.rootFieldItem)  
 const obj =TD.parse(strTemplate)
 
```

**Comment by jianwu** [comment]
You are right. Seems it's a real bug. I have reproduced this issue with following updated script run on https://npm.runkit.com/treedoc. Will debug and fix it. Thanks for the observation. 
```
var TD = require("treedoc").TD

this.rootFieldItem= {fieldName:"Root"}
this.rootFieldItem.arrChildFieldItem=[
          {fieldName:"Field 1", parent:this.rootFieldItem, fieldValue:"aa", fieldSize:5},
          {fieldName:"Field 2", parent:this.rootFieldItem, fieldValue:"bb", fieldSize:6},
          {fieldName:"Group 1", parent:this.rootFieldItem},
          {fieldName:"Field 3", parent:this.rootFieldItem, fieldValue:"cc", fieldSize:10}
]  
this.rootFieldItem.arrChildFieldItem[2].arrChildFieldItem=[
          { fieldName:"Field 4",parent:this.rootFieldItem.arrChildFieldItem[2],fieldValue:"cc1", fieldSize:5},
          {fieldName:"Field 5", parent:this.rootFieldItem.arrChildFieldItem[2],fieldValue:"cc2f", fieldSize:5}
]
const strTemplate = TD.stringify(this.rootFieldItem)  
obj =TD.parse(strTemplate)

```

**Comment by jianwu** [comment]
Issue has been fixed in: 0.3.24
